src/diamond/collector.py

```diff
@@ -56,8 +56,14 @@ def get_hostname(config, method=None):
                                     stdout=subprocess.PIPE)
             hostname = proc.communicate()[0].strip()
             if proc.returncode != 0:
-                raise subprocess.CalledProcessError(proc.returncode,
-                                                    config['hostname'])
+                skip_errors = config.get('hostname_cache_skip_errors')
+                # If we aren't explicitly skipping errors OR we have no cached
+                # result, raise an exception.
+                if not skip_errors or method not in get_hostname.cached_results:
+                    raise subprocess.CalledProcessError(proc.returncode,
+                                                        config['hostname'])
+                else:  # Recycle the previously cached version.
+                    hostname = get_hostname.cached_results[method][0]
```

Comment:
I think that you should at least log that this is happening somewhere. I see other diamond code that logs to a 'diamond' logger.

log = logging.getLogger('diamond')

If this were in another application I'd also suggest recording a metric for it so we can alert if it happens a lot, but I'm not sure how that would work on the diamond "manager" layer, especially when what you are doing is configuring how you log metrics in the first place (you don't want that failing).